src/Opdex.Platform.Application/EntryHandlers/Markets/GetMarketPermissionsForAddressQueryHandler.cs

```diff
@@ -0,0 +1,43 @@
+using MediatR;
+using Opdex.Platform.Application.Abstractions.EntryQueries.Markets;
+using Opdex.Platform.Application.Abstractions.Queries.Markets;
+using Opdex.Platform.Application.Abstractions.Queries.Markets.Permissions;
+using Opdex.Platform.Common.Enums;
+using Opdex.Platform.Common.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Opdex.Platform.Application.EntryHandlers.Markets
+{
+    public class GetMarketPermissionsForAddressQueryHandler : IRequestHandler<GetMarketPermissionsForAddressQuery, IEnumerable<MarketPermissionType>>
+    {
+        private readonly IMediator _mediator;
+
+        public GetMarketPermissionsForAddressQueryHandler(IMediator mediator)
+        {
+            _mediator = mediator ?? throw new ArgumentNullException(nameof(mediator));
+        }
+
+        public async Task<IEnumerable<MarketPermissionType>> Handle(GetMarketPermissionsForAddressQuery request, CancellationToken cancellationToken)
+        {
+            var market = await _mediator.Send(new RetrieveMarketByAddressQuery(request.Market, findOrThrow: true), cancellationToken);
+            if (market.IsStakingMarket) throw new NotFoundException("Market address must represent a standard market.");
+
+            var assignedPermissions = await _mediator.Send(new RetrieveMarketPermissionsByUserQuery(market.Id, request.Wallet), cancellationToken);
+            
+            var permissions = assignedPermissions.ToHashSet();
+
+            var isWalletMarketOwner = request.Wallet == market.Owner;
+
+            if (isWalletMarketOwner || !market.AuthPoolCreators) permissions.Add(MarketPermissionType.CreatePool);
+            if (isWalletMarketOwner || !market.AuthProviders) permissions.Add(MarketPermissionType.Provide);
+            if (isWalletMarketOwner || !market.AuthTraders) permissions.Add(MarketPermissionType.Trade);
```

Comment:
What about non market owners in markets that have permissions enabled 

I think it should be the following where we also pass through if the returned permissions includes one for the requesting user. 

```
if (isWalletMarketOwner || permissions.Any(p => p.Type == MarketPermissionType.CreatePool || !market.AuthPoolCreators)...
```